modules/datahandling/datahandler.go

```diff
@@ -25,33 +37,80 @@ func (dh DataHandler) Handle(wsID uint64, messageType int, message []byte) error
 		return err
 	}
 
-	// automatically delegates if the request is authenticated or not
-	fullR, err := getFullRequest(req)
+	// automatically determines if the request is authenticated or not
+	fullRequest, err := getFullRequest(req)
+
+	var closures []dhClosure
 
 	if err != nil {
-		utils.LogOnError(err, "Failed to construct full request")
-		return err
+		res := new(serverMessageWrapper)
+		res.Timestamp = time.Now().UnixNano()
+		res.Type = "Responce"
+		if err == ErrAuthenticationFailed {
+			utils.LogOnError(err, "User not logged in")
+			res.ServerMessage = response{
+				Status: unauthorized,
+				Tag:    req.Tag,
+				Data:   struct{}{}}
+		} else {
+			utils.LogOnError(err, "Failed to construct full request")
+			res.ServerMessage = response{
+				Status: unimplemented,
+				Tag:    req.Tag,
+				Data:   struct{}{}}
+		}
+		closures = []dhClosure{toSenderClosure{msg: res}}
+	} else {
+		closures, err = fullRequest.process(dh.Db)
+		if err != nil {
+			utils.LogOnError(err, "Failed to handle process request")
+		}
+	}
+
+	for _, closure := range closures {
```

Comment:
Are these for subscription/unsubscription? Could we also add comments as to document what this is for?